### ISSUE TITLE
How to build and use MIOpen with AMD Radeon RX 6700 XT (gfx1031)

### ISSUE THREAD

**Issue by junliume** [ROCmSoftwarePlatform/MIOpen#1309]
_issue opened on behalf of @aosewski_ 

[Question]
Understandably gfx1031 is not officially supported, while can the user experiment with it and MIOpen?

- [ ]  How to build MIOpen with AMD Radeon RX 6700 XT?
- [ ]  How to use MIOpen kernels and DBs 

```
# /opt/rocm/bin/rocminfo | grep gfx
  Name:                    gfx1031
      Name:                    amdgcn-amd-amdhsa--gfx1031
```

[Some Tests and Workarounds]

-  The default instruction for installation will be stuck at `GPU_DETECTION` stage, since `gfx1031` is not officially supported.
    - ~~One way is to use `cmake -DMIOPEN_BACKEND=HIPNOGPU`~~ (not recommended, see comment below: https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1309#issuecomment-982826860)
    -  Line `#139` in test/CMakeLists.txt seems to be too strict that everything other than officially supported would fail the build.
- **Can gfx1031 use kernels and databases created for gfx1030?**
    - If so, how can our users experiment with them? :)

**Comment by JehandadKhan** [comment]
Please note that `cmake -DMIOPEN_BACKEND=HIPNOGPU` would disable all GPU functionality in MIOpen and should not be used for this purpose.

> Can gfx1031 use kernels and databases created for gfx1030?

>    If so, how can our users experiment with them? :)

Once the build issue is addressed, the usual MIOpen variables to generate databases should work out of the box and need no additional code work. Namely, 

`MIOPEN_FIND_MODE=1` to generate the find database and `MIOPEN_FIND_ENFORCE=4` to tune kernels

**Comment by junliume** [comment]
_Carry over some questions from chat:_ They are valuable for every potential users of MIOpen from scratch :)

- [ ] I do not see any instructions on how to add and from where get the precompiled kernels database.

- [ ] Does this local tuning happens also when running tests?

- [ ] Are there any examples available which showcase how to use MIOpen?

For pre-compiled kernels it seems that we only support `gfx900`, `gfx906`, `gfx908`:

```
/devel/MIOpen# apt search miopenkernels
Sorting... Done
Full Text Search... Done
miopenkernels-gfx900-56kdb/Ubuntu 1.1.0.40500-56 amd64
  MIOpenKernels built using CMake

miopenkernels-gfx900-56kdb4.5.0/Ubuntu 1.1.0.40500-56 amd64
  MIOpenKernels built using CMake

miopenkernels-gfx900-64kdb/Ubuntu 1.1.0.40500-56 amd64
  MIOpenKernels built using CMake

miopenkernels-gfx900-64kdb4.5.0/Ubuntu 1.1.0.40500-56 amd64
  MIOpenKernels built using CMake

miopenkernels-gfx906-60kdb/Ubuntu 1.1.0.40500-56 amd64
  MIOpenKernels built using CMake

miopenkernels-gfx906-60kdb4.5.0/Ubuntu 1.1.0.40500-56 amd64
  MIOpenKernels built using CMake

miopenkernels-gfx906-64kdb/Ubuntu 1.1.0.40500-56 amd64
  MIOpenKernels built using CMake

miopenkernels-gfx906-64kdb4.5.0/Ubuntu 1.1.0.40500-56 amd64
  MIOpenKernels built using CMake

miopenkernels-gfx908-120kdb/Ubuntu 1.1.0.40500-56 amd64
  MIOpenKernels built using CMake

miopenkernels-gfx908-120kdb4.5.0/Ubuntu 1.1.0.40500-56 amd64
  MIOpenKernels built using CMake
```

**Comment by junliume** [comment]
> Please note that `cmake -DMIOPEN_BACKEND=HIPNOGPU` would disable all GPU functionality in MIOpen and should not be used for this purpose.
> 
> > Can gfx1031 use kernels and databases created for gfx1030?
> 
> > If so, how can our users experiment with them? :)
> 
> Once the build issue is addressed, the usual MIOpen variables to generate databases should work out of the box and need no additional code work. Namely,
> 
> `MIOPEN_FIND_MODE=1` to generate the find database and `MIOPEN_FIND_ENFORCE=4` to tune kernels

Indeed the user has the following issue with  `cmake -DMIOPEN_BACKEND=HIPNOGPU`

Any test throws an error `"hipErrorNoBinaryForGpu: Unable to find code object for all current devices!"`

**Comment by junliume** [comment]
Resolved by https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1531

**Comment by Looong01** [comment]
> Resolved by #1531

Sir, I have seen this issue.
Was gfx1031 supporting added to the develop branch of the official repo (https://github.com/ROCmSoftwarePlatform/MIOpen)? I have a 6700xt GPU and installed rocm 5.4.2. Now I want to install MIOpen. Can I install it by building MIOpen from source?

**Comment by atamazov** [comment]
@Looong01 please see updated answer at https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1531#issuecomment-1411946795
